docs/developers/Backend.md

```diff
@@ -0,0 +1,125 @@
+## Cromwell Backend Development
+
+### What’s a Backend?
+
+Cromwell is a workflow execution service, which means it takes a representation of work to do (tasks) as well as the
+dependencies and code flow between them. When it comes to actually executing the specific task, Cromwell delegates that
+work to a “backend”. A backend is therefore responsible for the actual execution of a single task on some underlying
+computing platform, such as Google Cloud Platform, Azure, AWS, TES, SLURM, etc. Backends are implemented as a software
```

Comment:
To save everyone future confusion, maybe we shouldn't include Azure and TES both in this list.